sailor/pai/utils.py

```diff
@@ -0,0 +1,63 @@
+"""Module for various utility functions, in particular those related to fetching data from remote oauth endpoints."""
+
+import logging
+import warnings
+
+from ..utils.oauth_wrapper import get_oauth_client
+from ..utils.config import SailorConfig
+from ..utils.utils import DataNotFoundWarning
+from ..assetcentral.utils import _compose_queries
+
+LOG = logging.getLogger(__name__)
+LOG.addHandler(logging.NullHandler())
+
+
+def _fetch_data(endpoint_url, unbreakable_filters=(), breakable_filters=()):
+    """Retrieve data from the pai service."""
+    filters = _compose_queries(unbreakable_filters, breakable_filters)
+    oauth_client = get_oauth_client('pai')
+
+    if not filters:
+        filters = ['']
+
+    result = []
+    for filter_string in filters:
+        params = {'$filter': filter_string} if filter_string else {}
+        params['$format'] = 'json'
+
+        endpoint_data = oauth_client.request('GET', endpoint_url, params=params)
+
+        if isinstance(endpoint_data, list):
+            result.extend(endpoint_data)
+        else:
+            result.append(endpoint_data)
+
+    if len(result) == 0:
+        warnings.warn(DataNotFoundWarning(), stacklevel=2)
+    return result
+
+
+def _pai_application_url():
+    """Return the Pai application URL from the SailorConfig."""
+    return SailorConfig.get('pai', 'application_url')
+
+
+class PaiEntity:
```

Comment:
I think this should be subclased from the `AssetcentralEntity` (or ideally from a new, common base-class).
Also need to think a little about names, since we're spelling out `Assetcentral` and using an acronym here... `PredictiveAssetInsightsEntity` is quite long, but may be necessary? Not sure...

Reply:
just let me know...

Reply:
I think `PredictiveAssetInsightsEntity` is okay. It won't be used in the spelt-out version anyway. only for tests maybe.

Reply:
I maybe should call also folder ( if not moved to AC) as predictiveAssetInsights ?

Reply:
done